From 32c805a290cc4a504516eeb29c8ebd15cabcd706 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 12 May 2020 20:11:14 +0530
Subject: [PATCH 01/10] octeontx2-af: Fix nix_inline_ipsec_cfg mailbox

commit ee106e71f590986cd420ab80aa14aaa5040bd95f from
git@git.assembla.com:cavium/WindRiver.linux.git

nix_inline_ipsec_cfg may be called by AF consumer
before any LF is attached to it hence call
rvu_get_blkaddr accordingly.

Fixes: 3bb5a5
("octeontx2-af: Return assigned NIX/CPT block address")

Change-Id: Ibf1e005f311ec4696c6aafe916cf079787aeecd1
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/28281
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 8fa751cdbb88..1dfaad0f1769 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -4378,7 +4378,7 @@ int rvu_mbox_handler_nix_inline_ipsec_cfg(struct rvu *rvu,
 	if (!is_block_implemented(rvu->hw, BLKADDR_CPT0))
 		return 0;
 
-	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_NIX, req->hdr.pcifunc);
+	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_NIX, 0);
 	if (blkaddr < 0)
 		return NIX_AF_ERR_AF_LF_INVALID;
 
-- 
2.17.1

